src/server/server_family.cc

```diff
@@ -1849,8 +1850,8 @@ void ServerFamily::ReplicaOfInternal(string_view host, string_view port_sv, Conn
   }
 
   // First, switch into the loading state
-  if (auto new_state = service_.SwitchState(GlobalState::ACTIVE, GlobalState::LOADING);
-      new_state != GlobalState::LOADING) {
+  if (auto [new_state, success] = service_.SwitchState(GlobalState::ACTIVE, GlobalState::LOADING);
+      success) {
```

Comment:
you forgot a negation here

Reply:
I think we shouldn't be checking here for success either. The code below can stop an _already_ running replica and change it, and we allow this even under the loading phase. So it should assert the final state